pkg/types/powervs/validation/machinepool.go

```diff
@@ -1,27 +1,84 @@
 package validation
 
 import (
-	"k8s.io/apimachinery/pkg/util/validation/field"
-
+	"github.com/google/uuid"
 	"github.com/openshift/installer/pkg/types/powervs"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"math"
+	"regexp"
 )
 
 // ValidateMachinePool checks that the specified machine pool is valid.
 func ValidateMachinePool(p *powervs.MachinePool, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-	/*
-		for i, zone := range p.Zones {
-			if !strings.HasPrefix(zone, platform.Region) {
-				allErrs = append(allErrs, field.Invalid(fldPath.Child("zones").Index(i), zone, fmt.Sprintf("Zone not in configured region (%s)", platform.Region)))
-			}
+
+	// Validate ServiceInstance
+	if !isUUID(p.ServiceInstance) {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("serviceinstance"), p.ServiceInstance, "Service Instance provided is not a UUID"))
+	}
+
+	// Validate Name
+	if !regexp.MustCompile(`^[a-zA-Z0-9-_]{1,}$`).MatchString(p.Name) {
+		allErrs = append(allErrs, field.Invalid(fldPath, p.Name, "Only letters (no accents), numbers, underscores and dashes are allowed"))
+	}
+	if len(p.Name) > 47 {
+		allErrs = append(allErrs, field.TooLong(fldPath, p.Name, 47))
+	}
+
+	// Validate VolumeIDs
+	for i, volumeID := range p.VolumeIDs {
+		if !isUUID(volumeID) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("volumeIDs").Index(i), volumeID, "Volume ID provided is not a UUID"))
 		}
+	}
 
-		if p.OSDisk.DiskType != "" {
-			diskTypes := sets.NewString("pd-standard", "pd-ssd")
-			if !diskTypes.Has(p.OSDisk.DiskType) {
-				allErrs = append(allErrs, field.NotSupported(fldPath.Child("diskType"), p.OSDisk.DiskType, diskTypes.List()))
-			}
-		}*/
+	// Validate Memory
+	// TODO(mjturek): Check maximums on processors and memory. The maximum is a dynamic value
+	//                tied to sys type, service instance, and availability. We'd need to query
+	//                for the current maximum to handle this properly.
```

Comment:
just curious: a) is there a query for this? and b) are you putting this into this pr or just a note for later?

Reply:
a) I don't know of a query for this. If it doesn't exist we'd need to ask powervs to expose it.
b) This is a note for later. Maybe a JIRA ticket would be preferred? I have this TODO habit from openstack and hadn't really considered that it might not be something done elsewhere.

Reply:
i asked around. i'll see what we find out. but yah let's put todo's either in our todos doc or in a jira task